### DISCUSSION TITLE
Use Datasyncclient with Blobs?

### DISCUSSION THREAD

**Discussion by ideas2appsaustralia** [Azure/azure-mobile-apps#344] [Ideas]
Hi Adrian

I've got azure blob storage for images, photos. I am wondering whether I can use the Datasyncclient and replicate whats done in this sample for tables with blobs instead? I'd use my own BlobController can just refer to a class on the backend API that has the blob code to down load them , but Ive just realised the DatasyncClientData classes are configured with Id fields etc that I won't have in the blob container so can't use getremotetable etc.

Would I be right, that I couldnt make use of Datasyncclient in this way?

How do mobile apps generally handles images? I like the idea of using an ASP.NET Core API backend because I can keep all azure Ids, secrets etc off the mobile client

Scott

**Comment by adrianhall** [comment]
Don't use DatasyncClient in this way.

Best way is to write your own equivalent.  When using Azure.Storage, the Last-Updated and Etag can be returned.  You can do a "download the deltas" from that.  You'd have to work out how to deal with queued uploads.

Makes me think I should do a separate mobile client that has data, files, push notifications, auth, etc.  It would be a very different package that the DatasyncClient, but I think it could work.
